lib/Dialect/RTG/Transforms/ElaborationPass.cpp

```diff
@@ -73,6 +74,15 @@ class ElaboratorValue {
     return isOpaque == other.isOpaque && value == other.value;
   }
 
+  virtual std::string toString() const {
+    std::string out;
+    llvm::raw_string_ostream stream(out);
+    stream << "<opaque ";
+    value.print(stream);
+    stream << " at " << this << ">";
+    return out;
+  }
```

Comment:
Can this be done by defining the `<<` operator. This can also, if only used in debug, can be guarded under `NDEBUG`. E.g., something like https://github.com/llvm/circt/blob/main/include/circt/Analysis/FIRRTLInstanceInfo.h#L198

Reply:
That's a great point! Thanks for reviewing!